django/forms/fields.py

```diff
@@ -135,11 +135,13 @@ def __init__(
         #             is its widget is shown in the form but not editable.
         # label_suffix -- Suffix to be added to the label. Overrides
         #                 form's label_suffix.
+        # bound_field_class -- BoundField class to use in Field.get_bound_field.
         self.required, self.label, self.initial = required, label, initial
         self.show_hidden_initial = show_hidden_initial
         self.help_text = help_text
         self.disabled = disabled
         self.label_suffix = label_suffix
+        self.bound_field_class = bound_field_class
```

Comment:
Refs this discussion https://github.com/django/django/pull/18266/files#r1810379643

I think we should add a `bound_field_class` on the `Field` so folk can subclass it and set it once. This line would then become:

```suggestion
        self.bound_field_class = bound_field_class or self.bound_field_class
```

This then mirrors the approach with forms which can be set when writing your form class or when instantiating it. 

We're missing this hook to be able to remove the recommendation from the docs to override `get_bound_field()`. As currently that's the only way folk can subclass a `Field` and just set it once and not have to remember to use their custom `bound_field_class` everywhere.

Reply:
Done. I also modified a test to asset that the `BoundField` passed to `Field`'s constructor takes priority over the `Field.bound_field_class`.

Reply:
Oh dear… Undocumening `get_bound_field` broke the docs of the 1.9 release. What should I do about it?

Reply:
> /home/runner/work/django/django/docs/releases/1.9.txt:366: WARNING: py:meth reference target not found: django.forms.Field.get_bound_field [ref.meth]

You can replace that reference in `1.9.txt` with just double backticks: 

<pre><code>``...``</code></pre>

Reply:
@christophehenry I think you've misunderstood what's  meant here though...

Earlier you added `Additionally it can override get_bound_field()` — that's not needed. (just point to the new method)

But we can't just un-document `get_bound_field()`. You should leave that how it was, maybe adding a link from there to the new attribute if you think that adds anything. 

The rest of the docs though can point (only) to the new attribute. 

Make sense?

Reply:
Wait, [this was earlier documented in `api.txt`](https://github.com/django/django/blob/main/docs/ref/forms/api.txt#L1453) in the section *Customizing ``BoundField``*. I think it shoud be moved to `fields.txt` now.